### PR TITLE
Add `Unschedulable` state to VM model

### PR DESCRIPTION
**Pull request by torchiaf**
#### PR Checklist
- Is this a multi-tenancy feature/bug?
    - [ ] Yes, the relevant RBAC changes are at:
- Do we need to backport changes to the [old Rancher UI](https://github.com/rancher/u), such as RKE1?
    - [ ] Yes, the relevant PR is at:
- Are backend engineers aware of UI changes?
    - [ ] Yes, the backend owner is:

Fixes https://github.com/harvester/harvester/issues/4982

### CODE REVIEW

File: pkg/harvester/models/kubevirt.io.virtualmachine.js [hunk 1]
```diff
@@ -632,6 +633,21 @@ export default class VirtVm extends HarvesterResource {
...
   }
 
+  get isUnschedulable() {
+    if (this.isBeingStopped) {
+      const condition = this.status?.conditions?.find(c => c.reason === UNSCHEDULABLE);
+
+      if (!!condition) {
+        return {
+          status:  UNSCHEDULABLE,
+          message: condition.message || '',
```
Comment (Yu-Jack) [inline pkg/harvester/models/kubevirt.io.virtualmachine.js:643]: Should we need to give the default message if `condition.message` is null to let users know what next step they could take?

**Review by Yu-Jack — COMMENTED** [review]
Seems well for the case stop it, then start it.

image

There is a question, should we consider change its label when it also have `Unschedulable` condition after starting?

image

File: pkg/harvester/models/kubevirt.io.virtualmachine.js (hunk 1 above)
Comment (torchiaf) [reply pkg/harvester/models/kubevirt.io.virtualmachine.js:643]: We can add a default message, but it's not actually used right now.

**Comment by torchiaf** [comment]
The `unschedulable` state should be visible even when VM is starting, thanks @Yu-Jack .

**Review by Yu-Jack — APPROVED** [review]
LGTM

image

**Review by bk201 — APPROVED** [review]

**Pull request merged**